### ISSUE TITLE
Add instructions for OS X Arm arch, M2 Macs with virtualbox to sig-windows-dev-tools README

### ISSUE THREAD

**Issue by pramitagautam** [kubernetes-sigs/sig-windows-dev-tools#218]
I have been following the README on https://github.com/kubernetes-sigs/sig-windows-dev-tools.
But the ubuntu control plane is not coming up but rather is always in aborted state.

```
bash-5.2$ vagrant status
cni:
calico
Current machine states:

controlplane              aborted (virtualbox)
winw1                     not created (virtualbox)
```
I tried bringing it up but its failing with the below error:-

```
bash-5.2$ vagrant up
cni:
calico
Bringing machine 'controlplane' up with 'virtualbox' provider...
Bringing machine 'winw1' up with 'virtualbox' provider...
==> controlplane: Checking if box 'roboxes/ubuntu2004' version '4.2.4' is up to date...
==> controlplane: Clearing any previously set forwarded ports...
==> controlplane: Clearing any previously set network interfaces...
==> controlplane: Preparing network interfaces based on configuration...
    controlplane: Adapter 1: nat
    controlplane: Adapter 2: hostonly
==> controlplane: Forwarding ports...
    controlplane: 22 (guest) => 2222 (host) (adapter 1)
==> controlplane: Running 'pre-boot' VM customizations...
==> controlplane: Booting VM...
==> controlplane: Waiting for machine to boot. This may take a few minutes...
    controlplane: SSH address: 127.0.0.1:2222
    controlplane: SSH username: vagrant
    controlplane: SSH auth method: private key
    controlplane: Warning: Connection reset. Retrying...
The guest machine entered an invalid state while waiting for it
to boot. Valid states are 'starting, running'. The machine is in the
'aborted' state. Please verify everything is configured
properly and try again.

If the provider you're using has a GUI that comes with it,
it is often helpful to open that and watch the machine, since the
GUI often has more helpful error messages than Vagrant can retrieve.
For example, if you're using VirtualBox, run `vagrant up` while the
VirtualBox GUI is open.

The primary issue for this error is that the provider you're using
is not properly configured. This is very rarely a Vagrant issue.
```

I have 1) arm 64 arch 2) MAC OS 3)I had to install Dev Preview beta version of virtualbox as the link says so :- https://www.virtualbox.org/wiki/Downloads
```
bash-5.2$ brew install --cask virtualbox
==> Downloading https://download.virtualbox.org/virtualbox/7.0.4/VirtualBox-7.0.4-154605-OSX.dmg
Already downloaded: /Users/gpramita/Library/Caches/Homebrew/downloads/f25294560ad1eeb0b6ad82e4551be058867a3e07a6e19e423c9871ad5028fce1--VirtualBox-7.0.4-154605-OSX.dmg
Error: Cask virtualbox depends on hardware architecture being one of [{:type=>:intel, :bits=>64}], but you are running {:type=>:arm, :bits=>64}.
```

**Comment by jayunit100** [comment]
So Longggg thread about this recently glad you rediscovered it so we can prioritize it. 

```
Hey, reviving 
[@Aravindh](https://kubernetes.slack.com/team/UHX473K4G)
 thread on https://kubernetes.slack.com/archives/C0SJ4AFB7/p1638563772223300
If someone is interested in help testing libvirt and KVM+Qemu in the sig-windows-dev-tools, I have converted the vbox image to qcow2 and published in the https://app.vagrantup.com/sig-windows-dev-tools/boxes/windows-2019
Tests were made using this Vagrantfile here with the --provider=libvirt - still experimenting with it..
AWS EC2 does not allow kvm, without this acceleration the VM is very slow.
https://github.com/kubernetes-sigs/sig-windows-dev-tools/issues/157#issuecomment-1329932332 :thread:




27 replies
Amim Knabben
  [8 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669684629735019?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
This could be a step in the long term (?) -
having CAPI specs running in kind can remove the bunch of scripts and let the bootstrap ctrl handle it
https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt
GitHubGitHub
[GitHub - kubernetes-sigs/cluster-api-provider-kubevirt: Cluster API Provider for KubeVirt](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt)
Cluster API Provider for KubeVirt. Contribute to kubernetes-sigs/cluster-api-provider-kubevirt development by creating an account on GitHub. (62 kB)
Amim Knabben
  [8 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669684747378309?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
In the short-term having the capability to choose the provider (vbox/qemu) is a good step..

PS C:\Users\vagrant> ipconfig

Windows IP Configuration


Ethernet adapter Ethernet 4:

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : fe80::7468:65b4:5dd5:7512%5
   IPv4 Address. . . . . . . . . . . : 192.168.121.116
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . : 192.168.121.1

Ethernet adapter Ethernet 5:

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : fe80::f115:e1fc:bf79:11e1%4
   IPv4 Address. . . . . . . . . . . : 10.20.30.135
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . : 10.20.30.1
Aravindh Puthiyaparambil
  [6 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669829587231959?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
That's awesome 
[@Amim Knabben](https://kubernetes.slack.com/team/U1UP4807N)
. I can help test it out.
Amim Knabben
  [6 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669829743459869?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
it's capi image with all on it, I would say it will work with kubevirt, didn't had too much time to go into
Amim Knabben
  [6 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669829757343019?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
if we have a cluster on kubevirt, it can be a good replacement for the project
Aravindh Puthiyaparambil
  [6 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669829770383749?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
Oh wow. That I do not have.
Amim Knabben
  [6 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669829793442059?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
maaaybe we can run the image on CAPK (https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt)
GitHubGitHub
[GitHub - kubernetes-sigs/cluster-api-provider-kubevirt: Cluster API Provider for KubeVirt](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt)
Cluster API Provider for KubeVirt. Contribute to kubernetes-sigs/cluster-api-provider-kubevirt development by creating an account on GitHub. (62 kB)
Aravindh Puthiyaparambil
  [6 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669829826742849?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
I was thinking we could use vagrant with libvirt for a local cluster
Amim Knabben
  [6 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669829842037649?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
yea, as a first step :slightly_smiling_face:
Amim Knabben
  [6 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669829860073589?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
if we replace the vbox by libvirt is a good improvement already
Aravindh Puthiyaparambil
  [6 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669829875737599?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
Yeah that's what I was going to try
:100:
1

Aravindh Puthiyaparambil
  [4 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670020954403359?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
So I started going down this path and am revisiting the getting vagrant to work with libvirt :cry:
sig-windows-dev-tools   master → origin ↑72 {3} ?:12  3  vagrant plugin install vagrant-libvirt
==> vagrant: A new version of Vagrant is available: 2.3.3 (installed version: 2.2.7)!
==> vagrant: To upgrade visit: https://www.vagrantup.com/downloads.html

Installing the 'vagrant-libvirt' plugin. This can take a few minutes...
Vagrant failed to properly resolve required dependencies. These
errors can commonly be caused by misconfigured plugin installations
or transient network issues. The reported error is:

racc requires Ruby version >= 2.5.
Aravindh Puthiyaparambil
  [4 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670020973155399?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
I vaguely remember having to run vagrant as a container to get around this
Aravindh Puthiyaparambil
  [4 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670025148584579?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
Fixed by updating to vagrant 2.3.3
Amim Knabben
  [3 days ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670067215749399?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
maybe upgrading your ruby package? is that a wsl2?
Aravindh Puthiyaparambil
  [1 day ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670257446788059?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
The issue was that my ruby version was 3.1.2 and the plugin needed >= 2.5 IOW it needed an older version of ruby. However upgrading vagrant resulted in a new version of the libvirt plugin being picked and that fixed things.
Amim Knabben
  [1 day ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670257565350799?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
nice
Aravindh Puthiyaparambil
  [1 day ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670257617471509?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
The other issue that I ran into and need to look at is how we are sharing folders in the VM
Amim Knabben
  [1 day ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670257941848909?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
it's required the vbguest plugin
Amim Knabben
  [1 day ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670257961527469?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
not sure how it works with libvirt :disappointed: or the equivalent (edited) 
Aravindh Puthiyaparambil
  [1 day ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670257991542199?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
It's different :wink: Uses NFS by default. I had to use rsync for Linux but not sure if that works with Windows.
:ack:
1

Amim Knabben
  [1 day ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670258015576839?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
oh
Aravindh Puthiyaparambil
  [1 day ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670262160889569?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
[Synced Folders](https://github.com/vagrant-libvirt/vagrant-libvirt/blob/main/docs/examples.markdown#synced-folders)
Amim Knabben
  [1 day ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670262187410219?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
nice
Amim Knabben
  [1 day ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670262229436959?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
maybe the virtio-fs is an option?
Aravindh Puthiyaparambil
  [24 hours ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670264868374949?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
Please do note that virtiofs depends on:
Host: Linux >= 5.4, QEMU >= 4.2 and Libvirt >= 6.2 (e.g. Ubuntu 20.10)
Guest: Linux >= 5.4 (e.g. Ubuntu 20.04)
No mention of Windows. AFAIK you need a virtio driver in the guest for this to work.
Aravindh Puthiyaparambil
  [24 hours ago](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1670265296094229?thread_ts=1669684319.649089&cid=C0SJ4AFB7)
[Installing virtiofs drivers on Windows](https://virtio-fs.gitlab.io/howto-windows.html)

```

**Comment by aravindhp** [comment]
sig-windows threads [[1](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1638563772223300),[2](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1669684319649089)] that discuss moving to vagrant-libvirt
